### PR TITLE
convert make_solution to async

### PR DESCRIPTION
**Pull request by geoffwalmsley**
Modify wallet_protocol.make_solution to async so it can accept an action_scope for vault.

### CODE REVIEW

File: chia/_tests/wallet/vault/test_vault_wallet.py [hunk 1]
```diff
@@ -185,16 +185,6 @@ async def test_vault_creation(
...
-    # test make_solution
-    coin = (await wallet.select_coins(uint64(100), DEFAULT_COIN_SELECTION_CONFIG)).pop()
-    wallet.make_solution(primaries, coin_id=coin.name())
-    with pytest.raises(ValueError):
-        wallet.make_solution(primaries)
-
-    # test match_hinted_coin
-    matched = await wallet.match_hinted_coin(wallet.vault_info.coin, wallet.vault_info.inner_puzzle_hash)
-    assert matched
-
```
Comment (Quexington) [inline chia/_tests/wallet/vault/test_vault_wallet.py:197]: Are these changes intentional?

Comment (geoffwalmsley) [reply chia/_tests/wallet/vault/test_vault_wallet.py:197]: yeah, those lines were just for coverage and won't be needed once the make_solution changes are done

Comment (Quexington) [reply chia/_tests/wallet/vault/test_vault_wallet.py:197]: Oh I see, I thought these were not vault tests, my bad.

**Review by Quexington — APPROVED** [review]

**Review by markelrod — APPROVED** [review]
